src/pytest_plugins/execute/eth_config/eth_config.py

```diff
@@ -142,7 +186,67 @@ def rpc_endpoint(request) -> str:
     return request.config.getoption("rpc_endpoint")
 
 
-@pytest.fixture(autouse=True, scope="session")
-def eth_rpc(rpc_endpoint: str) -> EthRPC:
-    """Initialize ethereum RPC client for the execution client under test."""
-    return EthRPC(rpc_endpoint)
+def all_rpc_endpoints(config) -> Dict[str, List[EthRPC]]:
+    """Derive a mapping of exec clients to the RPC URLs they are reachable at."""
+    rpc_endpoint = config.getoption("rpc_endpoint")
+    el_clients: List[str] = config.getoption("majority_clients")  # besu, erigon, ..
+    if len(el_clients) == 0:
+        return {}
+
+    pattern = r"(.*?@rpc\.)([^-]+)-([^-]+)(-.*)"
+    url_dict: Dict[str, List[EthRPC]] = {
+        exec_client: [
+            EthRPC(
+                re.sub(
+                    pattern,
+                    f"\\g<1>{consensus}-{exec_client}\\g<4>",
+                    rpc_endpoint,
+                )
+            )
+            for consensus in CONSENSUS_CLIENTS
+        ]
+        for exec_client in el_clients
+    }
+    # url_dict looks like this:
+    # {
+    #     'besu': [<EthRPC that holds url for grandine+besu>, <EthRPC that holds url for lighthouse+besu>, ..],  # noqa: E501
+    #     'erigon':  ...
+    #     ...
+    # }
+    return url_dict
+
+
+def pytest_generate_tests(metafunc: pytest.Metafunc):
+    """Generate tests for all clients under test."""
+    # all_rpc_endpoints is a dictionary with the name of the exec client as key
+    # and the possible URLs to contact it (different cl combinations) as value list
+    all_rpc_endpoints_dict = all_rpc_endpoints(metafunc.config)
+
+    if metafunc.definition.name == "test_eth_config_majority":
+        if len(all_rpc_endpoints_dict) < 2:
+            # The test function is not run because we only have a single client, so no majority comparison  # noqa: E501
+            logger.info(
+                "Skipping eth_config majority because less than 2 exec clients were passed"
+            )
+            metafunc.parametrize(
+                ["all_rpc_endpoints"],
+                [],
```

Comment:
```suggestion
                [
                    pytest.param(
                        all_rpc_endpoints_dict,
                        id=metafunc.definition.name,
                        marks=pytest.mark.skip("Only one client"),
                    )
                ],
```
Small fix so the test appears as "SKIPPED", along with the skip reason.